tensorboard/plugins/pr_curve/pr_curve_demo.py

```diff
@@ -0,0 +1,198 @@
+  
+  # Create a normal distribution layer used to generate true color labels.
+  channel_distribution = tf.distributions.Normal(loc=0., scale=42.)
+
+  # Sample the distribution to generate colors. Lets generate different numbers
+  # of each color. The first dimension is the count of examples.
+
+  # Generate reds.
+  number_of_reds = 100
+  true_reds = tf.clip_by_value(
+      tf.concat([
+          255 - tf.abs(channel_distribution.sample([number_of_reds, 1])),
+          tf.abs(channel_distribution.sample([number_of_reds, 2]))], axis=1),
+      0, 255)
+
+  # Generate greens.
+  number_of_greens = 200
+  true_greens = tf.clip_by_value(
+      tf.concat([
+          tf.abs(channel_distribution.sample([number_of_greens, 1])),
+          255 - tf.abs(channel_distribution.sample([number_of_greens, 1])),
+          tf.abs(channel_distribution.sample([number_of_greens, 1]))], axis=1),
+      0, 255)
+
+  # Generate blues.
+  number_of_blues = 150
+  true_blues = tf.clip_by_value(
+      tf.concat([
+          tf.abs(channel_distribution.sample([number_of_blues, 2])),
+          255 - tf.abs(channel_distribution.sample([number_of_blues, 1]))],
+          axis=1),
+      0, 255)
+
+  # Assign each color a vector of 3 booleans based on its true label.
+  labels = tf.concat([
+      tf.tile(tf.constant([[True, False, False]]), (number_of_reds, 1)),
+      tf.tile(tf.constant([[False, True, False]]), (number_of_greens, 1)),
+      tf.tile(tf.constant([[False, False, True]]), (number_of_blues, 1))
+  ], axis=0)
+
+  # We introduce 3 normal distributions. They are used to predict whether a
+  # color falls under a certain class (based on distances from corners of the
+  # color triangle). The distributions vary per color. We have the distributions
+  # narrow over time.
+  iteration = tf.Variable(0.0)
+  increment_iteration = tf.assign_add(iteration, 1.0)
+  with tf.control_dependencies([increment_iteration]):
+    red_predictor = tf.distributions.Normal(
+        loc=0., scale=tf.constant(40.) - iteration)
+    green_predictor = tf.distributions.Normal(
+        loc=0., scale=tf.constant(60.) - iteration)
+    blue_predictor = tf.distributions.Normal(
+        loc=0., scale=tf.constant(100.) - iteration)
+
+  # Make predictions (assign 3 probabilities to each color based on each color's
+  # distance to each of the 3 corners).
+  examples = tf.concat([true_reds, true_greens, true_blues], axis=0)
+  probabilities_colors_are_red = red_predictor.cdf(
+      tf.norm(examples - tf.constant([255., 0, 0]), axis=1))
+  probabilities_colors_are_green = green_predictor.cdf(
+      tf.norm(examples - tf.constant([0, 255., 0]), axis=1))
+  probabilities_colors_are_blue = blue_predictor.cdf(
+      tf.norm(examples - tf.constant([0, 0, 255.]), axis=1))
+  predictions = tf.concat([
+      tf.expand_dims(probabilities_colors_are_red, 1),
+      tf.expand_dims(probabilities_colors_are_green, 1),
+      tf.expand_dims(probabilities_colors_are_blue, 1)
```

Comment:
Huh, TIL. This is better than `tf.transpose(tf.stack([singleton]), perm=[1, 2, 0])`. Nice.